de.tu_bs.cs.isf.e4cf.core/src/de/tu_bs/cs/isf/e4cf/core/db/TableUtilities.java

```diff
@@ -0,0 +1,284 @@
+	 * @param tableName  String name of the table
+	 * @param columnName String name of the column
+	 * @return boolean
+	 * @throws SQLException
+	 */
+	public boolean columnExists(final String pPath, final String pDbName, final String tableName,
+			final String columnName) throws SQLException {
+		final Connection con = DatabaseFactory.getInstance().getDatabase(pPath, pDbName);
+		if (tableExists(pPath, pDbName, tableName)) {
+			final ResultSet rs = con.getMetaData().getColumns(null, null, tableName, null);
+			while (rs.next()) {
+				if (columnName.equals(rs.getString("COLUMN_NAME"))) {
+					con.close();
+					return true;
+				}
+			}
+		} else {
+			System.out.println("Table " + tableName + " does not exist.");
+		}
+		con.close();
+		return false;
+	}
+
+	/**
+	 * Method to get the primary key of a Table
+	 * 
+	 * @param pPath      String the path of the database
+	 * @param pDbName    String the name of the database
+	 * @param tableName  String name of the table
+	 * @param columnName String name of the column
+	 * @return List<String> list of the primary key in the Table
+	 * @throws SQLException
+	 */
+	public List<String> getPrimaryKeyTable(final String pPath, final String pDbName, final String tableName)
+			throws SQLException {
+		final Connection con = DatabaseFactory.getInstance().getDatabase(pPath, pDbName);
+		final ResultSet rs = con.getMetaData().getPrimaryKeys(null, null, tableName);
+		List<String> primarykeyList = new ArrayList<>();
+		while (rs.next()) {
+			String primarykey = rs.getString("COLUMN_NAME");
+			primarykeyList.add(primarykey);
+		}
+		return Collections.unmodifiableList(primarykeyList);
+	}
+
+	/**
+	 * Method to get the unique constraints of a Table
+	 * 
+	 * @param pPath      String the path of the database
+	 * @param pDbName    String the name of the database
+	 * @param tableName  String name of the table
+	 * @param columnName String name of the column
+	 * @return List<String> list of the unique keys in the Table
+	 * @throws SQLException
+	 */
+	public List<String> getUniqueKeyTable(final String pPath, final String pDbName, final String tableName)
+			throws SQLException {
+		final Connection con = DatabaseFactory.getInstance().getDatabase(pPath, pDbName);
+		final ResultSet rs = con.getMetaData().getIndexInfo(null, null, tableName, true, false);
+		List<String> uniquekeyList = new ArrayList<>();
+		while (rs.next()) {
+			String unique = rs.getString("COLUMN_NAME");
+			uniquekeyList.add(unique);
+		}
+		return Collections.unmodifiableList(uniquekeyList);
+	}
+
+	/**
+	 * Method to get column metadata
+	 *
+	 * @param pPath     String the path of the database
+	 * @param pDbName   String the name of the database
+	 * @param tableName String name of the table
+	 * @return List<String> list of the column metadata in the database
+	 * @throws SQLException
+	 */
+	public List<Column> getColumnsTable(final String pPath, final String pDbName, final String tableName)

```

Comment:
Weshalb sind viele variablen hier und auch in der DatabaseFactory final? Ist es nicht möglich z.B. Name und path einer Datenbank zu ändern etc.?